invenio_rdm_migrator/load/postgresql.py

```diff
@@ -0,0 +1,151 @@
+
+
+class PostgreSQLCopyLoad(Load):  # TODO: abstract SQL from PostgreSQL?
+    """PostgreSQL COPY load."""
+
+    def __init__(self, db_uri, table_loads, output_path):
+        """Constructor."""
+        self.db_uri = db_uri
+        self.output_dir = Path(output_path) / f"data/tables{_ts(iso=False)}"
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self._table_loads = table_loads
+
+    def _cleanup(self, db=False):
+        """Cleanup csv files and DB after load."""
+        for table in self.tables:
+            table.cleanup_files(db)
+
+    def _prepare(self, entries):
+        """Dump entries in csv files for COPY command."""
+
+        _prepared_tables = []
+        for table in self._table_loads:
+            # otherwise the generator is exahusted by the first table
+            # TODO: nested generators, how expensive is this copy op?
+            _prepared_tables.extend(
+                table.prepare(self.output_dir, entries=entries)
+            )
+        
+        return iter(_prepared_tables)  # yield at the end vs yield per table
+
+    def _load(self, table_entries):
+        """Bulk load CSV table files.
+
+        Loads the tables in the order given by the generator.
+        """
+        with psycopg.connect(self.db_uri) as conn:
+            for table in table_entries:
+                name = table._table_name
+                cols = ", ".join([f.name for f in fields(table)])
+                fpath = self.output_dir / f"{name}.csv"
+                file_size = fpath.stat().st_size  # total file size for progress logging
+
+                print(f"[{_ts()}] COPY FROM {fpath}")  # TODO: logging
+                with contextlib.ExitStack() as stack:
+                    cur = stack.enter_context(conn.cursor())
+                    copy = stack.enter_context(
+                        cur.copy(f"COPY {name} ({cols}) FROM STDIN (FORMAT csv)")
+                    )
+                    fp = stack.enter_context(open(fpath, "r"))
+
+                    block_size = 8192
+
+                    def _data_blocks(block_size=8192):
+                        data = fp.read(block_size)
+                        while data:
+                            yield data
+                            data = fp.read(block_size)
+
+                    for idx, block in enumerate(_data_blocks(block_size)):
+                        if idx % 100:
+                            cur_bytes = idx * block_size
+                            percentage = (cur_bytes / file_size) * 100
+                            progress = f"{cur_bytes}/{file_size} ({percentage:.2f}%)"
+                            print(f"[{_ts()}] {name}: {progress}")
+                        copy.write(block)
+                conn.commit()
+
+    def run(self, entries, cleanup=False):
+        """Load entries."""
+        table_entries = self._prepare(entries) 
+        self._load(table_entries)
+
+        if cleanup:
+            self._cleanup()
+
+
+class DBTableLoad(ABC):
```

Comment:
```suggestion
class TableGenerator(ABC):
```

minor: the `Load` suffix suggests that it's a subclass of `base.Load`. in reality, this class generates table rows/tuples... So any of the below (or even others) are good candidates:

- `Table(s)Generator`
- `Row(s)Generator`

Reply:
`TableGenerator` it is 🚀